File: 프로그래머스/lv2/64065. 튜플/튜플.py
from collections import Counter as ct


def solution(s):
    cnts = ct()
    strs = s.replace('{', '').replace('}}', '').split('},')
    for arr in strs:
        cnts += ct(arr.split(','))
    return list(map(lambda x: int(x[0]), cnts.most_common()))

File: 프로그래머스/lv2/64065. 튜플/튜플.kt
class Solution {
    val map = HashMap<Int, Int>()

    fun solution(s: String): IntArray {
        val split = s.replace("{", "").replace("}}", "").split("},")
        
        split.forEach { e ->
            e.split(',').forEach {
                val v = it.toInt()

                map[v] = (map[v] ?: 0) + 1
            }
        }

        return map.keys.sortedBy { -map[it]!! }.toIntArray()
    }
}